### PR TITLE
Subclasses in optimiser.

### LINKED ISSUE #7067: Add Subclass to Loadout Optimizer
For share-able builds, there's often a specific Subclass (and subclass configuration) associated with that build. Today this is often implicit in the choice of Exotic, but Darkness subclasses introduced mods-for-subclasses, which can be incorporated into both the total stats and the saved mods in the loadout. In the future, Light subclasses will get a similar treatment. We can probably skip supporting talent grid (subclass path) selection given that they will be reworked.

### PR DESCRIPTION
**Pull request by ryan-rushton**
This adds subclasses and their stats to the optimiser. 

It uses the aspect and fragment drawer currently used in the loadout drawer. This means it allows the user to pick their full suite of subclass options even though abilities have no effect on stats. Reasoning for this is we will save this subclass in loadouts and if we didn't they would have to save and edit the subclass to get their full setup, which sucks.

Closes #7067 

### PR's
- https://github.com/DestinyItemManager/DIM/pull/7566 **(this one)**
- https://github.com/DestinyItemManager/DIM/pull/7568 

Screen Shot 2021-12-27 at 10 09 44 pm

### CODE REVIEW

File: src/app/loadout-builder/filter/LockArmorAndPerks.tsx [hunk 1]
```diff
@@ -83,6 +91,27 @@ export default memo(function LockArmorAndPerks({
       } catch (e) {}
     };
 
+  const chooseSubclass = async () => {
```
Comment (bhollis) [inline src/app/loadout-builder/filter/LockArmorAndPerks.tsx:94]: Might be nice to share this function with the Loadouts drawer

File: src/app/loadout-builder/filter/LockArmorAndPerks.tsx [hunk 2]
```diff
@@ -123,6 +188,7 @@ export default memo(function LockArmorAndPerks({
           </ol>
         </div>
       )}
+      {/* Locked exotic */}
```
Comment (bhollis) [inline src/app/loadout-builder/filter/LockArmorAndPerks.tsx:191]: Might be nicer to just make them components

**Review by bhollis — APPROVED** [review]

File: src/app/loadout-builder/filter/LockArmorAndPerks.tsx (hunk 2 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/filter/LockArmorAndPerks.tsx:191]: Yeah planning on doing this in another PR. This component is really big now.

File: src/app/loadout-builder/filter/LockArmorAndPerks.tsx (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/filter/LockArmorAndPerks.tsx:94]: Can do

**Pull request merged**